ingest/exporter/ingestexportservice.py

```diff
@@ -436,43 +437,54 @@ def put_files_in_dss(self, bundle_uuid, files_to_put, process_info):
         created_files = []
 
         for bundle_file in files_to_put:
-            version = ''
-            file_uuid = bundle_file["dss_uuid"]
-            created_file = None
+            file_uuid = bundle_file.get('dss_uuid')
+            file_version = bundle_file.get('update_date')
             input_data_files = [input_file['dataFileUuid'] for input_file in list(process_info.input_files.values())]
 
             try:
                 # TODO if file is an input file, this file may already be in the data store, need to get the stored
-                #  version
-                # This assumes that the latest version is the file version in the input bundle, should be a safe
-                # assumption for now
-                # Ideally, bundle manifest must store the file uuid and version and version must be retrieved from there
+                #  version. This assumes that the latest version is the file version in the input bundle, should be a
+                #  safe assumption for now. Ideally, bundle manifest must store the file uuid and version and version
+                #  must be retrieved from there
 
                 # if metadata file , check is_from_input_bundle flag, if true, do not put file to DSS again
                 if bundle_file.get('is_from_input_bundle') or file_uuid in input_data_files:
-                    file_response = self.dss_api.head_file(bundle_file["dss_uuid"])
-                    created_file = {
-                        'version': file_response.headers['X-DSS-VERSION']
-                    }
+                    file_response = self.dss_api.head_file(uuid)
+                    file_version = file_response.headers['X-DSS-VERSION']
                 else:
-                    created_file = self.dss_api.put_file(bundle_uuid, bundle_file)
+                    file = self.check_file_in_dss(file_uuid, file_version)
```

Comment:
It's better to get the result of the head file at this point than just a bool , to prevent doing another request to get the file version if in case file is already in DSS. Currently, data file versions aren't being tracked in Ingest and if it already exists, the latest version is always being retrieved.